test/extern/ferguson/extern-where.chpl

```diff
@@ -0,0 +1,5 @@
+extern proc printf(fmt:c_string) where true;
```

Comment:
Since you didn't update tests for the changed error message, I think it would be good to add one locking in the behavior for exported functions.